discord/ext/commands/pagination.py

```diff
@@ -0,0 +1,58 @@
+import discord, asyncio
+from typing import List, Union
+
+from discord import abc
+from discord.ui.button import button
+
+class Paginate(discord.ui.View):
+    """Creates a paginator for embeds that is navigated with buttons.
+    
+    Parameters
+    ------------
+    send_to: :class:`discord.abc.Messageable`
+    embed: List[:class:`discord.Embed`]
+    """
+
+    def __init__(self, send_to, embeds: List[discord.Embed]):
+        super().__init__()
+        self.send_to = send_to
+        self.embeds = embeds
+        self.current_page = 1
+
+    @discord.ui.button(label = "<", style = discord.ButtonStyle.green, disabled=True)
+    async def previous(self, button: discord.ui.Button, interaction = discord.Interaction):
+        page_count = len(self.embeds)
+
+        if not self.current_page:
+            self.current_page = 1
+        else:
+            self.current_page -= 1
+
+        if self.current_page == 1:
+            button.disabled = True
+            self.children[1].disabled = False
+
+        await interaction.response.edit_message(embed = self.embeds[self.current_page-1], view = self)
+
+    @discord.ui.button(label = '>', style=discord.ButtonStyle.green)
+    async def forward(self, button: discord.ui.Button, interaction = discord.Interaction):
+        page_count = len(self.embeds)
+        self.children[0].disabled = False
+
+        if not self.current_page:
+            self.current_page = 1
+        else:
+            self.current_page += 1
+
+        if self.current_page == page_count:
+            button.disabled = True
+
+        await interaction.response.edit_message(embed = self.embeds[self.current_page-1], view = self)
+
+    async def send(self):
+        """Sends a message with the paginated embeds."""
+        if len(self.embeds) == 1:
+            self.children[1].disabled = True
```

Comment:
What if rather than disabling the pagination buttons, we didn't send them at all if they aren't needed? This feels like a bad UX, adding disabled buttons that will never be used.

Reply:
The issue though is that library development is that you need to provide something that is extensible, otherwise it won't be used. Sure, that's maybe how you'd build it, but someone else (like me) wouldn't use a builtin paginator if I paging buttons are shown when they have no use.

We need to cater to the masses, which is I think one reason why `discord.py` never added something like this outside of external extensions; it's too opinionated.

Reply:
I've added a `hide_disabled` parameter which does exactly what it sounds like.
Its just like the help command, its great to have for people who want to use it, but some wont like it and will want to do something else. With that said, the help command is also an extension so it makes sense for pagination to to be there.

Reply:
It looks to me like `hide_disabled` removes them after they're added, which would result in the button flashing in and out.

We might need to move beyond the builtin decorators and add items manually if we want to support a true option to hide buttons.

Reply:
I dont believe it works like that. It send the updated view after changes are made, it doesnt matter the order they are done. Also, I've obviously tested it and have had no occurrence of button flashing.

Reply:
Oh I see, we're removing it every single time before we send / edit. Hm.

Reply:
We could save a lot of headache by adding them conditionally.

```python
class MyView:
    def __init__(self, embeds, hide_disabled):
        if embed_count == 1 and not hide_disabled:
            # add next button
            # add previous button
        
    @property   
    def embed_count(self):
        ...
        
```

This way we only add them one time, and then can forget about removing / adding them later.

Reply:
Though `show_disabled` might might more sense in this case.

Reply:
I tried to take your advice on this, i changed it to show_disabled. 
As far as adding buttons goes, according to the d.py guys, there isnt an effective way to add the button if its needed and its better to just remove it if its not needed. But i did changed the way its done